File: full/res/tests/church.ml
let add1 = fun num: Nat (
  suc num
) in

let zero = fun f: Nat -> Nat (
  fun x: Nat (
    x
  )
) in 

let succ = fun n: 
  (Nat -> Nat) -> Nat -> Nat (
    fun f: Nat -> Nat (
      fun x: Nat (
        f (n f) x
      )
    )
) in 

let add3 = (succ succ succ zero) add1

in add3 0

File: full/res/tests/church.f
let add1 = func num: Nat (
    suc num
) in

let zero = func f0: Nat -> Nat (
    func x0: Nat (
        x0
    )
) in 

let succ = func nf: 
  (Nat -> Nat) -> Nat -> Nat (
    func ff: Nat -> Nat (
        func xf: Nat (
            ff ((nf ff) xf)
        )
    )
) in 

let add4 = (succ (succ (succ zero))) add1

in add4 Zero
      
